packages/app-desktop/gui/NoteEditor/utils/contextMenu.ts

```diff
@@ -53,17 +55,52 @@ function handleCopyToClipboard(options: ContextMenuOptions) {
 	}
 }
 
-export function menuItems(): ContextMenuItems {
+export async function openItemById(options: any, dispatch: Function) {
+	const itemId = options.resourceId || options.itemId;
+	const item = await BaseItem.loadItemById(itemId);
+
+	if (!item) throw new Error(`No item with ID ${itemId}`);
+
+	if (item.type_ === BaseModel.TYPE_RESOURCE) {
+		const localState = await Resource.localState(item);
+		if (localState.fetch_status !== Resource.FETCH_STATUS_DONE || !!item.encryption_blob_encrypted) {
+			if (localState.fetch_status === Resource.FETCH_STATUS_ERROR) {
+				bridge().showErrorMessageBox(`${_('There was an error downloading this attachment:')}\n\n${localState.fetch_error}`);
+			} else {
+				bridge().showErrorMessageBox(_('This attachment is not downloaded or not decrypted yet'));
+			}
+			return;
+		}
+
+		try {
+			await ResourceEditWatcher.instance().openAndWatch(item.id);
+		} catch (error) {
+			console.error(error);
+			bridge().showErrorMessageBox(error.message);
+		}
+	} else if (item.type_ === BaseModel.TYPE_NOTE) {
+		let noteItem: any = {
+			type: 'FOLDER_AND_NOTE_SELECT',
+			folderId: item.parent_id,
+			noteId: item.id,
+		};
+
+		if (options.hash) {
+			noteItem = { ...noteItem, hash: options.hash };
+		}
```

Comment:
Make no sense to recreate the object. Just pass the hash directly to the action, as in the original code.